### PR TITLE
Match check_for_ispy and check_for_pulse_stone

### PR DESCRIPTION
**Pull request by JoshDuMan**
check_for_pulse_stone - https://decomp.me/scratch/sBDIh
check_for_ispy - https://decomp.me/scratch/kVFfI

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.00%) on `us`.

⚠️ This PR introduces 4 warnings:
- src/77480.c:10: warning: redundant redeclaration of `D_8010C958' in same scope
- src/77480.c:219: warning: implicit declaration of function `update_player_input'
- src/77480.c:221: warning: implicit declaration of function `update_player_blink'
- src/77480.c:224: warning: implicit declaration of function `phys_update_action_state'
- src/77480.c:226: warning: implicit declaration of function `collision_main_lateral'
- src/77480.c:229: warning: implicit declaration of function `phys_update_standard'
- src/77480.c:229: warning: previous implicit declaration of `phys_update_standard'
- src/77480.c:231: warning: implicit declaration of function `phys_update_lava_reset'
- src/77480.c:231: warning: previous implicit declaration of `phys_update_lava_reset'
- src/77480.c:243: warning: implicit declaration of function `func_800EFD08'
- src/77480.c:246: warning: implicit declaration of function `func_800E0B90'
- src/77480.c:254: warning: implicit declaration of function `check_input_open_menus'
- src/77480.c:256: warning: implicit declaration of function `check_input_status_menu'
- src/77480.c:260: warning: implicit declaration of function `check_for_interactables'
- src/77480.c:261: warning: implicit declaration of function `check_for_conversation_prompt'
- src/77480.c:262: warning: implicit declaration of function `check_for_pulse_stone'
- src/77480.c:263: warning: implicit declaration of function `check_for_ispy'
- src/77480.c:263: warning: previous implicit declaration of `check_for_ispy'
- src/77480.c:286: warning: implicit declaration of function `partner_use_ability'
- src/77480.c:292: warning: `phys_update_standard' was previously implicitly declared to return `int'
- src/77480.c:292: warning: type mismatch with previous implicit declaration
- src/77480.c:301: warning: implicit declaration of function `phys_update_jump'
- src/77480.c:307: warning: implicit declaration of function `phys_update_falling'
- src/77480.c:311: warning: implicit declaration of function `check_input_midair_jump'
- src/77480.c:315: warning: implicit declaration of function `collision_check_player_overlaps'
- src/77480.c:322: warning: implicit declaration of function `func_800E4F10'
- src/77480.c:332: warning: implicit declaration of function `func_802BE070_31DBE0'
- src/77480.c:342: warning: `phys_update_lava_reset' was previously implicitly declared to return `int'
- src/77480.c:342: warning: type mismatch with previous implicit declaration
- src/77480.c:365: warning: implicit declaration of function `reset_player_status'
- src/77480.c:368: warning: implicit declaration of function `func_800E6B68'
- src/77480.c:373: warning: implicit declaration of function `func_800E5520'
- src/77480.c:390: warning: implicit declaration of function `get_overriding_player_anim'
- src/77480.c:498: warning: `check_for_ispy' was previously implicitly declared to return `int'
- src/77480.c:498: warning: type mismatch with previous implicit declaration
- src/77480.c:503: warning: passing arg 3 of `dma_copy' makes pointer from integer without a cast
- src/77480.c:514: warning: implicit declaration of function `func_802B7000_E225B0'
- src/77480.c:527: warning: implicit declaration of function `func_802B71D4'
- src/77480.c:553: warning: implicit declaration of function `func_802B71C8'
- src/77480.c:572: warning: implicit declaration of function `func_802B71E8_E202F8'
- src/77480.c:58: warning: implicit declaration of function `player_raycast_down'
- src/77480.c:6: warning: redundant redeclaration of `D_8010C920' in same scope
- src/77480.c:8: warning: redundant redeclaration of `D_8010C940' in same scope

File: src/77480.c [hunk 1]
```diff
@@ -1,14 +1,19 @@
 #include "common.h"
 #include "ld_addrs.h"
 
-extern s32 D_8010C920;
+#define E21870_VRAM_DEF 0x802B7000
+#define E225B0_VRAM_DEF 0x802B7000
```
Comment (nanaian) [inline src/77480.c:5]: It doesn't match to use ld_addrs.h, right?

Comment (ethteck) [reply src/77480.c:5]: yeah, it doesn't - talked to Josh about this earlier

**Review by ethteck — APPROVED** [review]

File: src/77480.c [hunk 2]
```diff
@@ -527,7 +522,43 @@ void func_800E0374(void) {
...
+        dy = gGameStatusPtr->mapID / 7;
+        if ((dx + dy) > 5) {
+            return;
+        }
+
+        if (!(gPlayerStatus.animFlags & (PLAYER_STATUS_ANIM_FLAGS_USING_PULSE_STONE | PLAYER_STATUS_ANIM_FLAGS_40))) {
+            return;
+        }
+
+        if (gPlayerStatus.flags & 0x20 || gPlayerStatus.statusMenuCounterinputEnabledCounter) {
```
Comment (Brotenko) [inline src/77480.c:548]: Please add a new flag to the enum in enums.h and use that one - Thanks!

**Review by Brotenko — CHANGES_REQUESTED** [review]
Sick work! Really cool to see how the pulse stone works

**Pull request merged**